feature/appc 1784 store/oem attribution on pay as guest

brunovarelaAptoide opened the pull request:
**What does this PR do?**

- This PR adds a new way to find the OEM id that is inserted in a apk. For that we added a external library.

**Database changed?**
 No

**Where should the reviewer start?**

- AdyenPaymentFragment.java
- OemIdExtractorService.java
- OemIdExtractorV2.java
- OemIdExtractorV1.java

**How should this be manually tested?**

Better talk to me about how we should test this since its not a simple flow.

Flow on how to test this or QA Tickets related to this use-case: [APPC-1784](https://aptoide.atlassian.net/browse/APPC-1784)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-1784](https://aptoide.atlassian.net/browse/APPC-1784)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.) 

Yes, extractor.jar


**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

brunovarelaAptoide committed:
[APPC-1784-Store/OEM attribution on Pay As Guest] Added the extractor

- Added the extractor.jar from the wallet

brunovarelaAptoide committed:
[APPC-1784-Store/OEM attribution on Pay As Guest] extractor working for injected OEM

- Added support for Kotlin since the extractor is built in Kotlin
- Created a ExtractorV2 that abstracts the extractor jar service
- Organize some code in AdyenPaymentFragment

brunovarelaAptoide committed:
[APPC-1784-Store/OEM attribution on Pay As Guest]  bugfix

- Fixed a bug from a || to and &&

brunovarelaAptoide committed:
[APPC-1784-Store/OEM attribution on Pay As Guest] Added extractor test class

- Test class to test the ExtractorService

brunovarelaAptoide committed:
[APPC-1784-Store/OEM attribution on Pay As Guest]  Unwanted Files

- Removed unwanted files

brunovarelaAptoide committed:
[APPC-1784-Store/OEM attribution on Pay As Guest] Removed one Log import

Trinkes commented on android-appcoins-billing/build.gradle:
```diff
@@ -96,19 +96,20 @@ android {
-  embed 'aptoide-client-v8:aptoide-analytics-core:1.0.4' //Needed so that if the dev use SDK the lib is automatically included
-
+  embed 'aptoide-client-v8:aptoide-analytics-core:1.0.4'
+  //Needed so that if the dev use SDK the lib is automatically included
```
why did you change this comment?

brunovarelaAptoide committed:
[APPC-1784-Store/OEM attribution on Pay As Guest] Remove formatting error

brunovarelaAptoide committed:
[APPC-1784-Store/OEM attribution on Pay As Guest] Remove formatting error

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/payasguest/OemIdExtractorV2.java:
```diff
@@ -0,0 +1,35 @@
+  
+  private String getApplicationDir(){
+    try {
+      return context.getPackageManager().getApplicationInfo(packageName,0).sourceDir;
```
you should use the package name passed as argument instead of getting it from contect

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/payasguest/OemIdExtractorV2.java:
```diff
@@ -0,0 +1,35 @@
+import com.aptoide.apk.injector.extractor.domain.IExtract;
+import java.io.File;
+
+public class OemIdExtractorV2 implements OemIdExtractor {
```
We need a better name for these classes and not use v1 and v2

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/service/address/OemIdExtractorService.java:
```diff
@@ -5,13 +5,18 @@
 
-  String extractOemId(String packageName) {
+  public String extractOemId(String packageName) {
+    String oemId = extractorV2.extract(packageName);
```
how do you know which extractor to give priority? In the class api is not clear 
I suggest to receive as constructor argument a list of sorted extractors and iterate them.

brunovarelaAptoide committed:
[APPC-1784-Store/OEM attribution on Pay As Guest] code review changes

- Added a list of extractor to be executed by the order of insertion in the list.
- Refactor class names.

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/payasguest/OemIdExtractorFromExternalLib.java:
```diff
@@ -0,0 +1,34 @@
+
+  public OemIdExtractorFromExternalLib(Context context) {
+    this.context = context;
+    this.extractor = new ExtractorV2();
```
why are you creating the `ExtractorV2` instance on constructor?

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/service/address/OemIdExtractorService.java:
```diff
@@ -1,18 +1,24 @@
+        return oemId;
+      }
+    }
+    return new String();
```
why do you return an empty string?how do you know it isn't a valid `oemId`?

brunovarelaAptoide committed:
[APPC-1784-Store/OEM attribution on Pay As Guest]Code review changes

- Instead of returning an empty string return null

brunovarelaAptoide committed:
[APPC-1784-Store/OEM attribution on Pay As Guest] Tests correction

brunovarelaAptoide committed:
[APPC-1784-Store/OEM attribution on Pay As Guest] Remove null sent to server

- If the oemid is null we wont send it to retrieve the adress
